### PR TITLE
Refactor RemoteImage API

### LINKED ISSUE #1468: `RemoteImage.registry` is populated only when a Neuro registry is used.
It gets really confusing when one instantiates `RemoteImage` manually and tries to populate constructor parameters. How would one know that if it is a Neuro registry, then `registry` should be populated, but when it is not, then the registry should go into `name`?

If everything goes to `name` then why keep `tag` separately?

### PR DESCRIPTION
**Pull request by anayden**
Closes #1468

### CODE REVIEW

File: neuromation/api/parsing_utils.py [hunk 1]
```diff
@@ -33,11 +56,13 @@ def __post_init__(self) -> None:
...
         if _is_in_neuro_registry(self):
-            name = f"https://{self.registry}/{self.owner}/{self.name}"
+            name = f"{self.registry}/{self.owner}/{self.name}"
             tag = f":{self.tag}" if self.tag else ""
             return name + tag
         else:
-            return str(self)
+            name = self.name
+            tag = f":{self.tag}" if self.tag else ""
+            return name + tag
```
Comment (serhiy-storchaka) [inline neuromation/api/parsing_utils.py:65]: It is common in both branches.

Comment (anayden) [reply neuromation/api/parsing_utils.py]: good point. I made the changes to simplify the code in the future and forgot about it. Updated

File: neuromation/api/parsing_utils.py [hunk 2]
```diff
@@ -19,6 +19,29 @@ class RemoteImage:
     registry: Optional[str] = None
     cluster_name: Optional[str] = None
 
+    @classmethod
+    def new_neuro_image(
```
Comment (serhiy-storchaka) [inline neuromation/api/parsing_utils.py:23]: In `new_external_image()` parameters are: `name`, `tag`, `registry` (the last two are optional).
In `new_neuro_image()` parameters are: `name`, `owner`, `registry`, `cluster_name`, `tag` (only `tag` is optional).
The order of arguments `tag` and `registry` is different, and there are other parameters inserted in the middle.

I propose to move `registry` after `name` and make all parameters except `name` and `registry` keyword-only. In both methods.

File: neuromation/api/parsing_utils.py [hunk 3]
```diff
@@ -32,12 +55,14 @@ def __post_init__(self) -> None:
...
-        else:
-            return str(self)
+        name = self.name
+        name = (
+            f"{self.registry}/{self.owner}/{name}"
+            if _is_in_neuro_registry(self)
+            else name
+        )
+        tag = f":{self.tag}" if self.tag else ""
+        return name + tag
```
Comment (serhiy-storchaka) [inline neuromation/api/parsing_utils.py:65]: Maybe so?
```suggestion
        name = self.name
        if _is_in_neuro_registry(self):
            name = f"{self.registry}/{self.owner}/{name}"
        if self.tag:
            name = f"{name}:{self.tag}"
        return name
```
Or so?
```suggestion
        prefix = f"{self.registry}/{self.owner}/" if _is_in_neuro_registry(self) else ""
        suffix = f":{self.tag}" if self.tag else ""
        return f"{prefix}{self.name}{suffix}"
```

File: docs/images_reference.rst [hunk 4]
```diff
@@ -189,5 +189,29 @@ RemoteImage
 
       URL that can be used to reference this image with Docker.
 
+    .. py:classmethod:: new_neuro_image(name: str, registry: str, *, owner: str, cluster_name: str, tag: Optional[str] = None) -> RemoteImage
+
+        Create a new instance referring to an image hosted on Neuro Platform
```
Comment (serhiy-storchaka) [inline docs/images_reference.rst:194]:
```suggestion
        Create a new instance referring to an image hosted on Neuro Platform.
```

File: docs/images_reference.rst [hunk 5]
```diff
@@ -189,5 +189,29 @@ RemoteImage
...
+
+      :param str owner: image owner name
+
+      :param str cluster_name: name of the cluster
+
+      :param str tag: image tag
+
+    .. py:classmethod:: new_external_image(name: str, registry: Optional[str] = None, *, tag: Optional[str] = None) -> RemoteImage
+
+        Create a new instance referring to an image hosted on an external registry (e.g. DockerHub)
```
Comment (serhiy-storchaka) [inline docs/images_reference.rst:208]:
```suggestion
        Create a new instance referring to an image hosted on an external registry (e.g. DockerHub_).
```

Comment (anayden) [reply docs/images_reference.rst:208]: I assume, you only suggested to add a dot at the end. Will commit this manually.

Comment (anayden) [reply docs/images_reference.rst:208]: Oh, nevermind, I figured it out.

**Pull request merged**